### PR TITLE
Added extra None message when the logger is stopped. This will break …

### PR DESCRIPTION
**Pull request by grandsuri**
…the loop without checking the queue contents and getting hung

### CODE REVIEW

File: dss_datamover/logger.py [hunk 1]
```diff
@@ -163,10 +162,15 @@ def logging(self, queue, lock, stop_logging):
                         (message_level, message_value) = message
                         print("{}: {}".format(LOGGING_LEVEL[message_level], message_value))
                         fh.write(str(time.ctime()) + ": " + LOGGING_LEVEL[message_level] + ": " + message_value + "\n")
+                    elif message is None:
+                        stop_flag = True
```
Comment (nsarras) [inline dss_datamover/logger.py:166]: question here. If the logging queue is ever empty, does that mean we want to stop the logger? Could it be possible for the queue to be empty at one point in time and then become populated at another time with more messages? Just wanted to verify this is the behavior we want in this case

File: dss_datamover/logger.py [hunk 2]
```diff
@@ -163,10 +162,15 @@ def logging(self, queue, lock, stop_logging):
...
+                        break
                     else:
                         fh.write(str(time.ctime()) + ": " + message + "\n")
 
-                if stop_logging.value and queue.qsize() == 0:
```
Comment (nsarras) [inline dss_datamover/logger.py:169]: We should also keep this condition. If stop_logging is set and queue is empty then we should also print a message and break here.

**Review by nsarras — CHANGES_REQUESTED** [review]
see added comments

File: dss_datamover/logger.py (hunk 2 above)
Comment (grandsuri) [reply dss_datamover/logger.py:169]: We are experiencing issues with MP queues. Even though there are no elements in queues, the queue calls are getting hung. Thats why we are doing a workaround.

File: dss_datamover/logger.py (hunk 1 above)
Comment (grandsuri) [reply dss_datamover/logger.py:166]: None message is the last one generated by the master_application/client_application after stopping all the processes (aka workers/tasks).

**Comment by nsarras** [comment]
ok thanks for explaining, we are good to go with this PR then.

**Review by nsarras — APPROVED** [review]

**Pull request merged**